components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/KeyStoreBasedInternalCryptoProvider.java

```diff
@@ -149,6 +164,33 @@ public KeyStoreBasedInternalCryptoProvider(KeyStore keyStore, String keyAlias, S
         }
     }
 
+    @Override
+    public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
+                          boolean returnSelfContainedCipherText) throws CryptoException {
+
+        byte[] encryptedKey = encrypt(cleartext, algorithm, javaSecurityAPIProvider);
+        String cipherTransformation = System.getProperty(CIPHER_TRANSFORMATION_SYSTEM_PROPERTY);
+        if (StringUtils.isNotBlank(cipherTransformation) && returnSelfContainedCipherText) {
```

Comment:
Check why StringUtils.isNotBlank(cipherTransformation) is specifically needed here?

Reply:
Fixed with 49e15cee8d90c5642c7f543629325e11e3ec6326